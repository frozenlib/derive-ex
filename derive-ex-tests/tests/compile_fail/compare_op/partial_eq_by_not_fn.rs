use derive_ex::derive_ex;

#[derive_ex(PartialEq)]
struct X {
    #[partial_eq(by = ())]
    x: u8,
}

fn main() {}
