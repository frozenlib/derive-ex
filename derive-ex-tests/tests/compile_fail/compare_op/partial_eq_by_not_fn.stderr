error[E0277]: expected a `Fn(&u8, &u8)` closure, found `()`
 --> tests/compile_fail/compare_op/partial_eq_by_not_fn.rs:5:23
  |
5 |     #[partial_eq(by = ())]
  |                       ^^ expected an `Fn(&u8, &u8)` closure, found `()`
6 |     x: u8,
  |     - required by a bound introduced by this call
  |
  = help: the trait `for<'a, 'b> Fn(&'a u8, &'b u8)` is not implemented for `()`
note: required by a bound in `__eq__x`
 --> tests/compile_fail/compare_op/partial_eq_by_not_fn.rs:3:1
  |
3 | #[derive_ex(PartialEq)]
  | ^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `__eq__x`
...
6 |     x: u8,
  |     - required by a bound in this function
  = note: this error originates in the attribute macro `derive_ex` (in Nightly builds, run with -Z macro-backtrace for more info)
